nvflare/app_common/executors/learner_executor.py

```diff
@@ -136,9 +191,116 @@ def validate(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal
         else:
             return make_reply(ReturnCode.EMPTY_RESULT)
 
+    @staticmethod
+    def _setup_learner2(learner: Learner2, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal):
```

Comment:
We should not have both _setup_learner() and _setup_learner2() in the same source code file. Maybe create a separate learner_executor2.py to fit the naming pattern?

Reply:
I thought about that. The only thing is that existing examples will have to change their config. I guess this might be okay...

Reply:
I think having a `Learner2Executor` might be more clean IF later we decide to deprecates Learner/LearnerExecutor.

Reply:
Good point. Will create a separate Learner2Executor and leave LearnerExecutor alone.